### PR TITLE
MRG: Hotfix for #812

### PR DESCRIPTION
**Pull request by hoechenberger**
In GH-812, I introduced a bug when trying to simplify things a little too much. I have corrected this now and expanded the respective comment in the code to explain
WTH we're doing there.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/read.py [hunk 1]
```diff
@@ -240,10 +240,15 @@ def _handle_scans_reading(scans_fname, raw, bids_path, verbose=False):
         if verbose:
             logger.debug(f'Loaded {scans_fname} scans file to set '
                          f'acq_time as {acq_time}.')
-        # Call anonymize() to remove any traces of the measurement date we wish
+        # First set measurement data to None and then call call anonymize() to
```
Comment (hoechenberger) [inline mne_bids/read.py:243]:
```suggestion
        # First set measurement date to None and then call call anonymize() to
```

**Review by agramfort — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
Closed & Re-opened to restart CI

**Comment by hoechenberger** [comment]
CI failing due to outdated caches. Why do we even cache pip stuff? It saves close to no time…
If anything, we need to make cache renewal more aggressive

**Comment by hoechenberger** [comment]
@sappelhoff I'm disabling the caches because it seems I keep ending up with on outdated version of MNE. Let's see if this works.

**Comment by hoechenberger** [comment]
@sappelhoff Re-enabling caching again … aaahh... what a ride 🙈

**Comment by sappelhoff** [comment]
Cis are the worst 😁

**Comment by hoechenberger** [comment]
> Cis are the worst 😁

German CIs are the Wurst!

**Comment by hoechenberger** [comment]
@sappelhoff Ok for you to merge?

**Comment by sappelhoff** [comment]
thanks for fixing this!

**Pull request merged**